test/pylib/manager_client.py

```diff
@@ -150,14 +150,16 @@ async def server_restart(self, server_id: ServerNum) -> None:
         await self.client.get_text(f"/cluster/server/{server_id}/restart")
         self._driver_update()
 
-    async def server_add(self, replace_cfg: Optional[ReplaceConfig] = None, cmdline: Optional[List[str]] = None) -> ServerInfo:
+    async def server_add(self, replace_cfg: Optional[ReplaceConfig] = None, cmdline: Optional[List[str]] = None, config: dict[str, str] = None) -> ServerInfo:
```

Comment:
One example of usefulness of enforcing type hints was me being sure that using `server_info.host_id` won't blow up in my face randomly at some point in CI because some code path ended up with `host_id = None`. Many tests depended on this field and assumed that it's never `None`. Now adding the possibility to only create servers but not start them means that we might sometimes not have access to the host ID. The types should reflect that so I can prepare for it when writing tests, for example. When I ask the question "can the host ID be missing?" I should be able to determine that from a single glance at the type instead of having to read the entire framework's code.

Here, in the example of our dict, I was actually hoping that you used `{}` instead of `None`, because 1. why use a larger type (`Optional[dict[str, str]]`) when you can use a smaller type (`dict[str, str]`), 2. each time we use `None` somewhere we also have to add a condition somewhere else to handle it (do this if not None else do that). If we used `dict[str, str]` the code would just merge it with the other config dict unconditionally, like it does with a non-empty config, without any `None`-`if`ology.

Reply:
> Many tests depended on this field and assumed that it's never None.

After reading the commits... that statement didn't age well.